presto-spark-base/src/main/java/com/facebook/presto/spark/execution/HttpNativeExecutionTaskInfoFetcher.java

```diff
@@ -41,26 +44,41 @@
 public class HttpNativeExecutionTaskInfoFetcher
 {
     private static final Logger log = Logger.get(HttpNativeExecutionTaskInfoFetcher.class);
+    private static final String TASK_ERROR_MESSAGE = "TaskInfoFetcher encountered too many errors talking to native process.";
 
     private final PrestoSparkHttpTaskClient workerClient;
     private final ScheduledExecutorService updateScheduledExecutor;
     private final AtomicReference<TaskInfo> taskInfo = new AtomicReference<>();
     private final Executor executor;
     private final Duration infoFetchInterval;
+    private final RequestErrorTracker errorTracker;
+    private final ScheduledExecutorService errorRetryScheduledExecutor;
+    private final AtomicReference<RuntimeException> lastException = new AtomicReference<>();
 
     @GuardedBy("this")
     private ScheduledFuture<?> scheduledFuture;
 
     public HttpNativeExecutionTaskInfoFetcher(
             ScheduledExecutorService updateScheduledExecutor,
+            ScheduledExecutorService errorRetryScheduledExecutor,
             PrestoSparkHttpTaskClient workerClient,
             Executor executor,
-            Duration infoFetchInterval)
+            Duration infoFetchInterval,
+            Duration maxErrorDuration)
     {
         this.workerClient = requireNonNull(workerClient, "workerClient is null");
         this.updateScheduledExecutor = requireNonNull(updateScheduledExecutor, "updateScheduledExecutor is null");
         this.executor = requireNonNull(executor, "executor is null");
         this.infoFetchInterval = requireNonNull(infoFetchInterval, "infoFetchInterval is null");
+        this.errorRetryScheduledExecutor = requireNonNull(errorRetryScheduledExecutor, "errorRetryScheduledExecutor is null");
+        this.errorTracker = new RequestErrorTracker(
+                "NativeExecution",
+                workerClient.getLocation(),
+                NATIVE_EXECUTION_TASK_ERROR,
+                TASK_ERROR_MESSAGE,
+                maxErrorDuration,
+                errorRetryScheduledExecutor,
+                "getting native process status");
```

Comment:
This message doesn't seem right. I assume we are getting task's info, not status of the native process.